gnuradio-runtime/include/gnuradio/sys_paths.h

```diff
@@ -10,18 +11,51 @@
 #ifndef GR_SYS_PATHS_H
 #define GR_SYS_PATHS_H
 
+#include "api.h"
 #include <gnuradio/api.h>
+#include <filesystem>
 
 namespace gr {
+namespace paths {
+/*! \brief directory to create temporary files.
+ *
+ * On UNIX-oid systems, typically /tmp.
+ */
+GR_RUNTIME_API std::filesystem::path tmp();
```

Comment:
Does the name change provide enough value to bother people with deprecation warnings?

Reply:
yes. The main point is that the return value changes, from a manually (actually, leaking or use after free or concurrently modified in the current API) `const char*` to a `std::filesystem::path`.